nilearn/_utils/niimg_conversions.py

```diff
@@ -369,8 +369,7 @@ def check_niimgs(niimgs, accept_3d=False, return_iterator=False):
     if (dim + depth) != 4:
         # Detailed error message that tells exactly the user what
         # was provided and what should have been provided.
-        raise TypeError("Data must be either a 4D Nifti image or a"
-                        " list of 3D Nifti images. You provided a %s%dD"
+        raise TypeError("Data must a be Niimg-like object. You provided a %s%dD"
```

Comment:
I think you need to say 4D Niimg-like object (and probably add the URL so that the error is helpful)